File: Intermediate/22.Write a program to evaluate a postfix expression/Solution.py
def evaluate_postfix(expression):
    stack = []
    for ch in expression:
        if ch.isdigit():
            stack.append(int(ch))
        else:
            val2 = stack.pop()
            val1 = stack.pop()
            if ch == '+':
                stack.append(val1 + val2)
            elif ch == '-':
                stack.append(val1 - val2)
            elif ch == '*':
                stack.append(val1 * val2)
            elif ch == '/':
                stack.append(val1 // val2)
            elif ch == '^':
                stack.append(val1 ** val2)
    return stack.pop()

if __name__ == "__main__":
    expression = input("Enter postfix expression: ")
    result = evaluate_postfix(expression)
    print(f"Result: {result}")

File: Intermediate/22.Write a program to evaluate a postfix expression/Solution.java
import java.util.Scanner;
import java.util.Stack;

public class PostfixEvaluator {

    public static int evaluatePostfix(String expression) {
        Stack<Integer> stack = new Stack<>();

        for (char ch : expression.toCharArray()) {
            if (Character.isDigit(ch)) {
                stack.push(ch - '0');
            } else {
                int val2 = stack.pop();
                int val1 = stack.pop();
                switch (ch) {
                    case '+': stack.push(val1 + val2); break;
                    case '-': stack.push(val1 - val2); break;
                    case '*': stack.push(val1 * val2); break;
                    case '/': stack.push(val1 / val2); break;
                    case '^': stack.push((int)Math.pow(val1, val2)); break;
                }
            }
        }
        return stack.pop();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter postfix expression: ");
        String expression = sc.next();

        int result = evaluatePostfix(expression);
        System.out.println("Result: " + result);
    }
}
